### PR TITLE
(Fixes #369, Fixes #380, Closes #425, Closes #392) Make SparkCluster and SparkStep compatible to EMR release label 4.x

### LINKED ISSUE #369: Issues using EMR4.3.0
I have a bunch of AWS DataPipelines using EMR4.3.0 with Spark 1.6.0 and naturally I want to programatically define them rather than manually writing up JSON definition files.

I just recently came across the Hyperion project which seems like something I could use. Just by looking at the Spark example and a bit of digging around though, it seems that the library is just supporting older style of data pipelines definitions. 

For example the apply methods of both SparkTaskActivity and SparkStep are instantiating their private companion classes with hard coded values for:
scriptRunner = "s3://elasticmapreduce/libs/script-runner/script-runner.jar",
jobRunner = s"${hc.scriptUri}run-spark-step.sh",

These are pre EMR 4.x.x ways of running spark. In EMR4.x.x it's recommended to use command-runner.jar and spark-submit e.g.:
```
{
 "name":"MyActivity",
 "step":[
  "command-runner.jar,",
  "spark-submit,",
  "--deploy-mode,cluster,",
  ...
 ]
 ...
}
```

I also noticed that in SparkCluster the way to specify Spark is via a standard bootstrap action s3://support.elasticmapreduce/spark/install-spark. The EMR 4.x.x way is to specify "applications":["spark"] in the EMRCluster object. 

There are other differences as well.

Some questions:
- Do you guys have plans to upgrade the DSL to support new EMR versions? When?
- Currently the configs I mentioned above are hard coded behind private case classes which cannot be directly instantiated by code using the library. Am I missing something. Is there another way to extend the functionality to allow generation of definitions for EMR4.x.x? Or do I need to modify the Hyperion code to do that?
- If modifying the Hyperion code is the only way forward, would you guys be willing to accept contributions in this direction?

### LINKED ISSUE #380: Make SparkCluster and SparkStep compatible to EMR release label 4.x
Currently spark cluster in using bootstrap actions to install spark, which worked fine in AMI 3.x, but not in release label 4.x. It needs to be updated to be able to work with release label 4.4 through configuration and change SparkStep accordingly.

### PR DESCRIPTION
**Pull request by sethyates**

### CODE REVIEW

File: core/src/main/scala/com/krux/hyperion/resource/SparkCluster.scala [hunk 1]
```diff
@@ -13,7 +13,7 @@ case class SparkCluster private (
   baseFields: BaseFields,
   resourceFields: ResourceFields,
   emrClusterFields: EmrClusterFields,
-  sparkVersion: HString
+  sparkVersion: Option[HString]
```
Comment (realstraw) [inline core/src/main/scala/com/krux/hyperion/resource/SparkCluster.scala]: If I understand correctly, this field is only used for non-release label clusters?

**Comment by realstraw** [comment]
build.sbt is not updated with the new version

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,10 @@
 All notable changes to this project will be documented in this file.
 This project adheres to [Semantic Versioning](http://semver.org/).
 
+## 3.5.3 - 2016-07-09
```
Comment (realstraw) [inline CHANGELOG.md]: 3.6.0 maybe?

File: core/src/main/scala/com/krux/hyperion/resource/SparkCluster.scala (hunk 1 above)
Comment (sethyates) [inline core/src/main/scala/com/krux/hyperion/resource/SparkCluster.scala]: Yes AFAICT

File: CHANGELOG.md (hunk 2 above)
Comment (sethyates) [inline CHANGELOG.md]: Indeed

**Comment by realstraw** [comment]
👍

**Pull request merged**